fms/utils/generation.py

```diff
@@ -53,17 +148,25 @@ def generate(
         use_cache: requires that the model accept use_cache and
             past_key_value_states args in forward method.
     """
-    batched = False
     if num_beams != 1:
         raise NotImplementedError("generate() does yet not support beam search")
-    if type(input_ids) == torch.Tensor:
-        if input_ids.dim() != 1:
-            batched = True
-    else:
-        raise RuntimeError("generate() requires a tensor of token ids as the prefix")
 
-    if not batched:
-        input_ids = input_ids.unsqueeze(0)
+    # a mapping that contains kwargs that are model specific
+    model_specific_kwargs: MutableMapping[str, Any] = dict()
```

Comment:
this is position_ids and mask right? so specific to padding, but not really specific to a particular model? maybe `masking_kwargs` or `padding_kwargs` ?

Reply:
updated in latest commit